fixed findDupes, added demo of findDupes running on artemisData FASTA files (repbase)

ttriche opened the pull request:
Needs some manual merging and/or oversight, kicking the can to Anthony to review :-)

Works pretty damned well in conjunction with collapseTpm(kexp, biotype_class="whatever").

arcolombo commented:
ok. add this to the list.  need to finish up the data variance review too... 




> On Sep 26, 2015, at 4:28 PM, Tim Triche, Jr. <notifications@github.com> wrote:
> 
> Needs some manual merging and/or oversight, kicking the can to Anthony to review :-)
> 
> Works pretty damned well in conjunction with collapseTpm(kexp, biotype_class="whatever").
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/RamsinghLab/TxDbLite/pull/11
> 
> Commit Summary
> 
> Update README.md
> Update README.md
> Update README.md
> Update README.md
> Update README.md
> Update README.md
> Automatically merge Repbase FASTA names in stubs
> erccDbLite package support, toss out some changes
> resolve some naming issues
> various fixes, updates, and a typo
> fixed issues with organism name abbreviations
> get rid of underscores
> tidy up createAnnotationPackage for artemis
> updates & simplifications
> replace hard-coded organism information with a data.frame
> fixed bug in getAnnotationType
> add support for GC content and remove old cruft
> Fix proper naming of metadata/organism at creation
> fix organism assignment in repDbLite creation
> fix a(nother stupid new) bug in ensDbLiteFromFasta
> transcriptsBy(gene, promoter, tx_biotype, gene_biotype, biotype_class)
> Add genesBy, improve transcriptsBy, genes, transcripts, etc.
> File Changes
> 
> M DESCRIPTION (4)
> M NAMESPACE (9)
> M R/Methods.R (124)
> D R/addBiotypeClasses.R (24)
> D R/ensDbLiteExonsFromGtf.R (52)
> M R/ensDbLiteFromFasta.R (79)
> M R/erccDbLiteFromFasta.R (4)
> A R/getGCcontent.R (33)
> M R/makeEnsDbLitePkg.R (32)
> A R/makeErccDbLitePkg.R (54)
> M R/makeRepDbLitePkg.R (34)
> M R/repDbLiteFromFasta.R (19)
> M R/utils.R (84)
> M README.md (22)
> A data/supportedOrganismsForTxDbLite.rda (0)
> A inst/erccdblite/DESCRIPTION (16)
> A inst/erccdblite/NAMESPACE (9)
> A inst/erccdblite/R/zzz.R (18)
> A inst/erccdblite/man/package.Rd (40)
> M man/TxDbLite-class.Rd (30)
> D man/addBiotypeClasses.Rd (15)
> D man/ensDbLiteExonsFromGtf.Rd (17)
> M man/ensDbLiteFromFasta.Rd (5)
> A man/getGCcontent.Rd (29)
> M man/makeEnsDbLitePkg.Rd (4)
> A man/makeErccDbLitePkg.Rd (27)
> M man/makeRepDbLitePkg.Rd (4)
> M man/utils.Rd (20)
> Patch Links:
> 
> https://github.com/RamsinghLab/TxDbLite/pull/11.patch
> https://github.com/RamsinghLab/TxDbLite/pull/11.diff
> —
> Reply to this email directly or view it on GitHub.
>

arcolombo commented:
I've pulled the changes from triche-master and merged on my local branch.  will test and see if it completes on my end, then git push the changes.

ttriche commented:
obvious solution: shitcan pathview and keggrest because they are a pain,
and use Reactome all the time everywhere.

--t

On Wed, May 4, 2016 at 11:39 AM, Anthony R. Colombo <
notifications@github.com> wrote:

> The travis builds are a pain because they are not stable. These builds
> using the current .yml passed. now they fail because pathview is not being
> installed before keggrest which imports it. then why didn't this fail last
> week/ my arcolombo branch is passing. need to pass RL with same changes
> 
> ## building RL github repo.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/RamsinghLab/TxDbLite/pull/11
> Commit Summary
> - travis pathview must be first b4 KEGGREST
> 
> File Changes
> - _M_ .travis.yml
>   https://github.com/RamsinghLab/TxDbLite/pull/11/files#diff-0 (4)
> 
> Patch Links:
> - https://github.com/RamsinghLab/TxDbLite/pull/11.patch
> - https://github.com/RamsinghLab/TxDbLite/pull/11.diff
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> https://github.com/RamsinghLab/TxDbLite/pull/11

arcolombo commented:
great solution.    I will roll this out after I finish the workflow  with
reactome enrichment... need to swat bugs...  updates...

On Wed, May 4, 2016 at 12:33 PM, Tim Triche, Jr. notifications@github.com
wrote:

> obvious solution: shitcan pathview and keggrest because they are a pain,
> and use Reactome all the time everywhere.
> 
> --t
> 
> On Wed, May 4, 2016 at 11:39 AM, Anthony R. Colombo <
> notifications@github.com> wrote:
> 
> > The travis builds are a pain because they are not stable. These builds
> > using the current .yml passed. now they fail because pathview is not
> > being
> > installed before keggrest which imports it. then why didn't this fail
> > last
> > week/ my arcolombo branch is passing. need to pass RL with same changes
> > 
> > ## building RL github repo.
> > 
> > You can view, comment on, or merge this pull request online at:
> > 
> > https://github.com/RamsinghLab/TxDbLite/pull/11
> > Commit Summary
> > - travis pathview must be first b4 KEGGREST
> > 
> > File Changes
> > - _M_ .travis.yml
> >   https://github.com/RamsinghLab/TxDbLite/pull/11/files#diff-0 (4)
> > 
> > Patch Links:
> > - https://github.com/RamsinghLab/TxDbLite/pull/11.patch
> > - https://github.com/RamsinghLab/TxDbLite/pull/11.diff
> > 
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly or view it on GitHub
> > https://github.com/RamsinghLab/TxDbLite/pull/11
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/RamsinghLab/TxDbLite/pull/11#issuecomment-216975735

arcolombo commented:
This has an important change, and TRAVIS will be offline for a day or so... I will santize it once the workflow is written and Phase II beta is up.

here is proof that it works for user input... it is long... but the short story is **reactomeSets must return ensembl  IDs  which match \* orignal rownames(kexp).  splitting up the list caused duplicate naming conventions which had to be detected and plucked.    

> reactomeSets("Mus musculus","gene")->t

lapply(t,nchar)
$`R-MMU-1368071`
[1] 18

$`R-MMU-1368082`
[1] 18 18 18 18 18

$`R-MMU-1368092`
[1] 18 18 18 18 18 18

$`R-MMU-1368108`
 [1] 18 18 18 18 18 18 18 18 18 18 18 18 18

$`R-MMU-1368110`
 [1] 18 18 18 18 18 18 18 18 18 18 18 18 18

$`R-MMU-1912408`
[1] 18

$`R-MMU-1989781`
[1] 18

$`R-MMU-2122947`
[1] 18 18 18

$`R-MMU-2197563`
[1] 18 18 18

$`R-MMU-2426168`
[1] 18 18

$`R-MMU-400253`
[1] 18 18 18

$`R-MMU-5617472`
 [1] 18 18 18 18 18 18 18 18 18 18 18 18

$`R-MMU-5667092`
[1] 18 18 18 18 18 18

$`R-MMU-8851708`
[1] 18

> reactomeSets("Mus musculus","transcript")
> checking if the cache ensembl names have uniform lengths ...
> done

$`R-MMU-1912408`
[1] "ENSMUST00000028288"

$`R-MMU-5617472`
 [1] "ENSMUST00000000964" "ENSMUST00000014848" "ENSMUST00000019117"
 [4] "ENSMUST00000047793" "ENSMUST00000049241" "ENSMUST00000055334"
 [7] "ENSMUST00000100375" "ENSMUST00000100523" "ENSMUST00000101395"
[10] "ENSMUST00000111980" "ENSMUST00000111983" "ENSMUST00000114434"

> reactomeSets("Mus musculus","transcript")->t
> checking if the cache ensembl names have uniform lengths ...
> done
> 
> lapply(t,nchar)
> $`R-MMU-1912408`
> [1] 18

$`R-MMU-5617472`
 [1] 18 18 18 18 18 18 18 18 18 18 18 18


reactomeSets("Homo sapiens","transcript")->t
checking if the cache ensembl names have uniform lengths ...
I detected duplicates, correcting the dupe names...
I split the reactomeSet, now correcting the ID lengths for duplicates...
done

> lapply(t,nchar)
> $`R-HSA-1257604`
> [1] 15

$`R-HSA-1799339`
[1] 15 15

$`R-HSA-1912408`
[1] 15 15 15 15

$`R-HSA-2559580`
[1] 15 15

$`R-HSA-2559585`
[1] 15 15

$`R-HSA-381042`
[1] 15

$`R-HSA-381070`
[1] 15 15

$`R-HSA-4086398`
[1] 15 15

$`R-HSA-428359`
 [1] 15 15 15 15 15 15 15 15 15 15

$`R-HSA-452723`
[1] 15

$`R-HSA-4641265`
[1] 15

$`R-HSA-5368286`
[1] 15 15 15

$`R-HSA-5389840`
[1] 15 15 15

$`R-HSA-5419276`
[1] 15 15 15

$`R-HSA-5607761`
[1] 15 15

$`R-HSA-5617472`
 [1] 15 15 15 15 15 15 15 15 15 15 15 15

> reactomeSets("Homo sapiens","gene")->t

 lapply(t,nchar)
$`R-HSA-111448`
[1] 15

$`R-HSA-1234158`
[1] 15 15 15

$`R-HSA-1368071`
[1] 15 15 15 15 15 15

$`R-HSA-1368082`
[1] 15 15 15 15 15

$`R-HSA-1368108`
 [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15

$`R-HSA-139915`
[1] 15

$`R-HSA-156711`
[1] 15 15 15 15 15

$`R-HSA-1912408`
[1] 15 15 15 15 15 15 15

$`R-HSA-195253`
[1] 15 15

######NOW for user input.....

 reactomeSets("Mus musculus","gene",mp)->t
 creating reactome set from mapped reactome list...
I detected duplicates...correcting...
I split the reactomeSet, now correcting the ID lengths for duplicates...
done

> t
> $`R-MMU-1368082`
> [1] "ENSMUSG00000020538" "ENSMUSG00000020889" "ENSMUSG00000024900"
> [4] "ENSMUSG00000026077" "ENSMUSG00000055116"

$`R-MMU-1368092`
[1] "ENSMUSG00000020538" "ENSMUSG00000020889" "ENSMUSG00000024900"
[4] "ENSMUSG00000026077" "ENSMUSG00000029238" "ENSMUSG00000055116"

$`R-MMU-1368108`
 [1] "ENSMUSG00000020038" "ENSMUSG00000020572" "ENSMUSG00000020889"
 [4] "ENSMUSG00000020893" "ENSMUSG00000022383" "ENSMUSG00000030103"
 [7] "ENSMUSG00000030256" "ENSMUSG00000031443" "ENSMUSG00000032238"
[10] "ENSMUSG00000055866" "ENSMUSG00000059824" "ENSMUSG00000068742"

$`R-MMU-1368110`
 [1] "ENSMUSG00000020038" "ENSMUSG00000020572" "ENSMUSG00000020889"
 [4] "ENSMUSG00000020893" "ENSMUSG00000022383" "ENSMUSG00000030103"
 [7] "ENSMUSG00000030256" "ENSMUSG00000031443" "ENSMUSG00000032238"
[10] "ENSMUSG00000055866" "ENSMUSG00000059824" "ENSMUSG00000068742"

$`R-MMU-1912408`
[1] "ENSMUSG00000026923"

$`R-MMU-1989781`
[1] "ENSMUSG00000022383"

$`R-MMU-2122947`
[1] "ENSMUSG00000019789" "ENSMUSG00000040289"

$`R-MMU-2197563`
[1] "ENSMUSG00000022528" "ENSMUSG00000048001"

$`R-MMU-2426168`
[1] "ENSMUSG00000024978" "ENSMUSG00000041220"

$`R-MMU-400253`
[1] "ENSMUSG00000029238" "ENSMUSG00000032238" "ENSMUSG00000055116"

$`R-MMU-5617472`
 [1] "ENSMUSG00000000942" "ENSMUSG00000014704" "ENSMUSG00000018973"
 [4] "ENSMUSG00000029844" "ENSMUSG00000038692" "ENSMUSG00000042448"
 [7] "ENSMUSG00000048763" "ENSMUSG00000075394" "ENSMUSG00000075588"
[10] "ENSMUSG00000079277" "ENSMUSG00000079560" "ENSMUSG00000101174"

$`R-MMU-5667092`
[1] "ENSMUSG00000020889" "ENSMUSG00000026077" "ENSMUSG00000029167"
[4] "ENSMUSG00000029238" "ENSMUSG00000055116"

$`R-MMU-8851708`
[1] "ENSMUSG00000030849"

> lapply(t,nchar)
> $`R-MMU-1368082`
> [1] 18 18 18 18 18

$`R-MMU-1368092`
[1] 18 18 18 18 18 18

$`R-MMU-1368108`
 [1] 18 18 18 18 18 18 18 18 18 18 18 18

$`R-MMU-1368110`
 [1] 18 18 18 18 18 18 18 18 18 18 18 18

$`R-MMU-1912408`
[1] 18

$`R-MMU-1989781`
[1] 18

$`R-MMU-2122947`
[1] 18 18

$`R-MMU-2197563`
[1] 18 18

$`R-MMU-2426168`
[1] 18 18

$`R-MMU-400253`
[1] 18 18 18

$`R-MMU-5617472`
 [1] 18 18 18 18 18 18 18 18 18 18 18 18

$`R-MMU-5667092`
[1] 18 18 18 18 18

$`R-MMU-8851708`
[1] 18

$`R-HSA-1989781`
 [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
[26] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15

$`R-HSA-201722`
[1] 15 15

$`R-HSA-2032785`
[1] 15 15

$`R-HSA-2122947`
[1] 15 15 15 15 15 15

$`R-HSA-2151201`
 [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15

$`R-HSA-2173796`
[1] 15 15 15 15 15

$`R-HSA-2197563`
[1] 15 15 15 15

$`R-HSA-2426168`
 [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15

$`R-HSA-2559580`
[1] 15 15 15 15 15

$`R-HSA-2559582`
[1] 15 15 15 15 15 15

$`R-HSA-2559584`
[1] 15

$`R-HSA-2559585`
[1] 15 15

$`R-HSA-2559586`
[1] 15

$`R-HSA-2644606`
[1] 15 15 15 15 15 15

$`R-HSA-2892245`
[1] 15 15 15 15 15 15 15

$`R-HSA-2892247`
[1] 15 15 15 15 15 15 15 15

$`R-HSA-2894862`
[1] 15 15 15 15 15 15

$`R-HSA-3371453`
 [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
[26] 15 15 15 15 15 15

$`R-HSA-3371511`
 [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
[26] 15 15 15 15 15 15

$`R-HSA-3371568`
 [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
[26] 15 15 15 15 15 15

$`R-HSA-3371571`
 [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
[26] 15 15 15 15 15 15

$`R-HSA-380994`
[1] 15 15 15 15 15 15 15

$`R-HSA-381038`
 [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
[26] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15

$`R-HSA-381183`
[1] 15 15 15 15 15

$`R-HSA-381340`
 [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15

$`R-HSA-381771`
[1] 15

$`R-HSA-400253`
[1] 15 15 15 15 15

$`R-HSA-400511`
[1] 15

$`R-HSA-4086398`
[1] 15

$`R-HSA-422085`
[1] 15

$`R-HSA-427389`
[1] 15

$`R-HSA-427413`
[1] 15

$`R-HSA-4411364`
[1] 15 15

$`R-HSA-452723`
[1] 15 15 15 15 15

$`R-HSA-4641265`
[1] 15

$`R-HSA-5467343`
[1] 15

$`R-HSA-5467345`
[1] 15

$`R-HSA-5578768`
[1] 15

$`R-HSA-5607761`
[1] 15 15

$`R-HSA-5610787`
[1] 15 15 15

$`R-HSA-5617472`
 [1] 15 15 15 15 15 15 15 15 15 15 15

$`R-HSA-5625886`
[1] 15 15

$`R-HSA-5628897`
[1] 15 15 15 15 15 15 15 15 15

$`R-HSA-5635851`
[1] 15 15 15 15 15

$`R-HSA-5660668`
[1] 15

$`R-HSA-5663220`
[1] 15

$`R-HSA-5683826`
[1] 15 15 15 15 15 15

$`R-HSA-5687128`
[1] 15 15 15 15 15 15 15 15

$`R-HSA-6796648`
 [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15

$`R-HSA-6803204`
 [1] 15 15 15 15 15 15 15 15 15 15

$`R-HSA-6803205`
[1] 15 15 15 15 15 15 15

$`R-HSA-6803207`
[1] 15 15 15 15 15 15

$`R-HSA-6803211`
[1] 15 15 15 15 15 15

$`R-HSA-6803529`
[1] 15

$`R-HSA-6804114`
[1] 15 15 15

$`R-HSA-6804115`
[1] 15 15 15 15 15

$`R-HSA-6804116`
[1] 15 15 15 15 15

$`R-HSA-6804754`
[1] 15 15

$`R-HSA-6804757`
[1] 15 15

$`R-HSA-6804759`
[1] 15

$`R-HSA-6807505`
[1] 15 15 15 15 15 15 15

$`R-HSA-69895`
[1] 15

$`R-HSA-73762`
[1] 15

$`R-HSA-73863`
[1] 15

$`R-HSA-8849473`
[1] 15

$`R-HSA-8849474`
[1] 15

$`R-HSA-8851708`
[1] 15


$`R-HSA-5628897`
[1] 15

$`R-HSA-5660668`
[1] 15

$`R-HSA-5687128`
[1] 15 15

$`R-HSA-6784531`
[1] 15

$`R-HSA-6785470`
 [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
[26] 15 15 15 15 15 15 15 15 15 15 15 15

$`R-HSA-6796648`
[1] 15 15 15 15 15 15 15 15

$`R-HSA-6803529`
[1] 15 15

$`R-HSA-6804116`
[1] 15

$`R-HSA-6807505`
[1] 15 15 15

$`R-HSA-69895`
[1] 15

$`R-HSA-72165`
[1] 15 15 15

$`R-HSA-72312`
 [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
[26] 15 15 15

$`R-HSA-8857538`
[1] 15

arcolombo commented:
I detect a duplicate by finding the original lengths  of rownames(kexp) , or in cache,  then compare with the lengths of allTerms$ID after the split,  If the split names have a length of 1 larger than the original name, then it is a duplicate and I substring out the last char....

ttriche commented:
We can't just use unique() somewhere in there?  Oh well 

--t

> On May 4, 2016, at 5:40 PM, Anthony R. Colombo notifications@github.com wrote:
> 
> This has an important change, and TRAVIS will be offline for a day or so... I will santize it once the workflow is written and Phase II beta is up.
> 
> here is proof that it works for user input... it is long... but the short story is **reactomeSets must return ensembl IDs which match \* orignal rownames(kexp). splitting up the list caused duplicate naming conventions which had to be detected and plucked.
> 
> reactomeSets("Mus musculus","gene")->t
> 
> lapply(t,nchar)
> $R-MMU-1368071
> [1] 18
> 
> $R-MMU-1368082
> [1] 18 18 18 18 18
> 
> $R-MMU-1368092
> [1] 18 18 18 18 18 18
> 
> $R-MMU-1368108
> [1] 18 18 18 18 18 18 18 18 18 18 18 18 18
> 
> $R-MMU-1368110
> [1] 18 18 18 18 18 18 18 18 18 18 18 18 18
> 
> $R-MMU-1912408
> [1] 18
> 
> $R-MMU-1989781
> [1] 18
> 
> $R-MMU-2122947
> [1] 18 18 18
> 
> $R-MMU-2197563
> [1] 18 18 18
> 
> $R-MMU-2426168
> [1] 18 18
> 
> $R-MMU-400253
> [1] 18 18 18
> 
> $R-MMU-5617472
> [1] 18 18 18 18 18 18 18 18 18 18 18 18
> 
> $R-MMU-5667092
> [1] 18 18 18 18 18 18
> 
> $R-MMU-8851708
> [1] 18
> 
> reactomeSets("Mus musculus","transcript")
> checking if the cache ensembl names have uniform lengths ...
> done
> 
> $R-MMU-1912408
> [1] "ENSMUST00000028288"
> 
> $R-MMU-5617472
> [1] "ENSMUST00000000964" "ENSMUST00000014848" "ENSMUST00000019117"
> [4] "ENSMUST00000047793" "ENSMUST00000049241" "ENSMUST00000055334"
> [7] "ENSMUST00000100375" "ENSMUST00000100523" "ENSMUST00000101395"
> [10] "ENSMUST00000111980" "ENSMUST00000111983" "ENSMUST00000114434"
> 
> reactomeSets("Mus musculus","transcript")->t
> checking if the cache ensembl names have uniform lengths ...
> done
> 
> lapply(t,nchar)
> $R-MMU-1912408
> [1] 18
> 
> $R-MMU-5617472
> [1] 18 18 18 18 18 18 18 18 18 18 18 18
> 
> reactomeSets("Homo sapiens","transcript")->t
> checking if the cache ensembl names have uniform lengths ...
> I detected duplicates, correcting the dupe names...
> I split the reactomeSet, now correcting the ID lengths for duplicates...
> done
> 
> lapply(t,nchar)
> $R-HSA-1257604
> [1] 15
> 
> $R-HSA-1799339
> [1] 15 15
> 
> $R-HSA-1912408
> [1] 15 15 15 15
> 
> $R-HSA-2559580
> [1] 15 15
> 
> $R-HSA-2559585
> [1] 15 15
> 
> $R-HSA-381042
> [1] 15
> 
> $R-HSA-381070
> [1] 15 15
> 
> $R-HSA-4086398
> [1] 15 15
> 
> $R-HSA-428359
> [1] 15 15 15 15 15 15 15 15 15 15
> 
> $R-HSA-452723
> [1] 15
> 
> $R-HSA-4641265
> [1] 15
> 
> $R-HSA-5368286
> [1] 15 15 15
> 
> $R-HSA-5389840
> [1] 15 15 15
> 
> $R-HSA-5419276
> [1] 15 15 15
> 
> $R-HSA-5607761
> [1] 15 15
> 
> $R-HSA-5617472
> [1] 15 15 15 15 15 15 15 15 15 15 15 15
> 
> reactomeSets("Homo sapiens","gene")->t
> 
> lapply(t,nchar)
> $R-HSA-111448
> [1] 15
> 
> $R-HSA-1234158
> [1] 15 15 15
> 
> $R-HSA-1368071
> [1] 15 15 15 15 15 15
> 
> $R-HSA-1368082
> [1] 15 15 15 15 15
> 
> $R-HSA-1368108
> [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
> 
> $R-HSA-139915
> [1] 15
> 
> $R-HSA-156711
> [1] 15 15 15 15 15
> 
> $R-HSA-1912408
> [1] 15 15 15 15 15 15 15
> 
> $R-HSA-195253
> [1] 15 15
> 
> ######NOW for user input.....
> 
> reactomeSets("Mus musculus","gene",mp)->t
> creating reactome set from mapped reactome list...
> I detected duplicates...correcting...
> I split the reactomeSet, now correcting the ID lengths for duplicates...
> done
> 
> t
> $R-MMU-1368082
> [1] "ENSMUSG00000020538" "ENSMUSG00000020889" "ENSMUSG00000024900"
> [4] "ENSMUSG00000026077" "ENSMUSG00000055116"
> 
> $R-MMU-1368092
> [1] "ENSMUSG00000020538" "ENSMUSG00000020889" "ENSMUSG00000024900"
> [4] "ENSMUSG00000026077" "ENSMUSG00000029238" "ENSMUSG00000055116"
> 
> $R-MMU-1368108
> [1] "ENSMUSG00000020038" "ENSMUSG00000020572" "ENSMUSG00000020889"
> [4] "ENSMUSG00000020893" "ENSMUSG00000022383" "ENSMUSG00000030103"
> [7] "ENSMUSG00000030256" "ENSMUSG00000031443" "ENSMUSG00000032238"
> [10] "ENSMUSG00000055866" "ENSMUSG00000059824" "ENSMUSG00000068742"
> 
> $R-MMU-1368110
> [1] "ENSMUSG00000020038" "ENSMUSG00000020572" "ENSMUSG00000020889"
> [4] "ENSMUSG00000020893" "ENSMUSG00000022383" "ENSMUSG00000030103"
> [7] "ENSMUSG00000030256" "ENSMUSG00000031443" "ENSMUSG00000032238"
> [10] "ENSMUSG00000055866" "ENSMUSG00000059824" "ENSMUSG00000068742"
> 
> $R-MMU-1912408
> [1] "ENSMUSG00000026923"
> 
> $R-MMU-1989781
> [1] "ENSMUSG00000022383"
> 
> $R-MMU-2122947
> [1] "ENSMUSG00000019789" "ENSMUSG00000040289"
> 
> $R-MMU-2197563
> [1] "ENSMUSG00000022528" "ENSMUSG00000048001"
> 
> $R-MMU-2426168
> [1] "ENSMUSG00000024978" "ENSMUSG00000041220"
> 
> $R-MMU-400253
> [1] "ENSMUSG00000029238" "ENSMUSG00000032238" "ENSMUSG00000055116"
> 
> $R-MMU-5617472
> [1] "ENSMUSG00000000942" "ENSMUSG00000014704" "ENSMUSG00000018973"
> [4] "ENSMUSG00000029844" "ENSMUSG00000038692" "ENSMUSG00000042448"
> [7] "ENSMUSG00000048763" "ENSMUSG00000075394" "ENSMUSG00000075588"
> [10] "ENSMUSG00000079277" "ENSMUSG00000079560" "ENSMUSG00000101174"
> 
> $R-MMU-5667092
> [1] "ENSMUSG00000020889" "ENSMUSG00000026077" "ENSMUSG00000029167"
> [4] "ENSMUSG00000029238" "ENSMUSG00000055116"
> 
> $R-MMU-8851708
> [1] "ENSMUSG00000030849"
> 
> lapply(t,nchar)
> $R-MMU-1368082
> [1] 18 18 18 18 18
> 
> $R-MMU-1368092
> [1] 18 18 18 18 18 18
> 
> $R-MMU-1368108
> [1] 18 18 18 18 18 18 18 18 18 18 18 18
> 
> $R-MMU-1368110
> [1] 18 18 18 18 18 18 18 18 18 18 18 18
> 
> $R-MMU-1912408
> [1] 18
> 
> $R-MMU-1989781
> [1] 18
> 
> $R-MMU-2122947
> [1] 18 18
> 
> $R-MMU-2197563
> [1] 18 18
> 
> $R-MMU-2426168
> [1] 18 18
> 
> $R-MMU-400253
> [1] 18 18 18
> 
> $R-MMU-5617472
> [1] 18 18 18 18 18 18 18 18 18 18 18 18
> 
> $R-MMU-5667092
> [1] 18 18 18 18 18
> 
> $R-MMU-8851708
> [1] 18
> 
> $R-HSA-1989781
> [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
> [26] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
> 
> $R-HSA-201722
> [1] 15 15
> 
> $R-HSA-2032785
> [1] 15 15
> 
> $R-HSA-2122947
> [1] 15 15 15 15 15 15
> 
> $R-HSA-2151201
> [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
> 
> $R-HSA-2173796
> [1] 15 15 15 15 15
> 
> $R-HSA-2197563
> [1] 15 15 15 15
> 
> $R-HSA-2426168
> [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
> 
> $R-HSA-2559580
> [1] 15 15 15 15 15
> 
> $R-HSA-2559582
> [1] 15 15 15 15 15 15
> 
> $R-HSA-2559584
> [1] 15
> 
> $R-HSA-2559585
> [1] 15 15
> 
> $R-HSA-2559586
> [1] 15
> 
> $R-HSA-2644606
> [1] 15 15 15 15 15 15
> 
> $R-HSA-2892245
> [1] 15 15 15 15 15 15 15
> 
> $R-HSA-2892247
> [1] 15 15 15 15 15 15 15 15
> 
> $R-HSA-2894862
> [1] 15 15 15 15 15 15
> 
> $R-HSA-3371453
> [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
> [26] 15 15 15 15 15 15
> 
> $R-HSA-3371511
> [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
> [26] 15 15 15 15 15 15
> 
> $R-HSA-3371568
> [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
> [26] 15 15 15 15 15 15
> 
> $R-HSA-3371571
> [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
> [26] 15 15 15 15 15 15
> 
> $R-HSA-380994
> [1] 15 15 15 15 15 15 15
> 
> $R-HSA-381038
> [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
> [26] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
> 
> $R-HSA-381183
> [1] 15 15 15 15 15
> 
> $R-HSA-381340
> [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
> 
> $R-HSA-381771
> [1] 15
> 
> $R-HSA-400253
> [1] 15 15 15 15 15
> 
> $R-HSA-400511
> [1] 15
> 
> $R-HSA-4086398
> [1] 15
> 
> $R-HSA-422085
> [1] 15
> 
> $R-HSA-427389
> [1] 15
> 
> $R-HSA-427413
> [1] 15
> 
> $R-HSA-4411364
> [1] 15 15
> 
> $R-HSA-452723
> [1] 15 15 15 15 15
> 
> $R-HSA-4641265
> [1] 15
> 
> $R-HSA-5467343
> [1] 15
> 
> $R-HSA-5467345
> [1] 15
> 
> $R-HSA-5578768
> [1] 15
> 
> $R-HSA-5607761
> [1] 15 15
> 
> $R-HSA-5610787
> [1] 15 15 15
> 
> $R-HSA-5617472
> [1] 15 15 15 15 15 15 15 15 15 15 15
> 
> $R-HSA-5625886
> [1] 15 15
> 
> $R-HSA-5628897
> [1] 15 15 15 15 15 15 15 15 15
> 
> $R-HSA-5635851
> [1] 15 15 15 15 15
> 
> $R-HSA-5660668
> [1] 15
> 
> $R-HSA-5663220
> [1] 15
> 
> $R-HSA-5683826
> [1] 15 15 15 15 15 15
> 
> $R-HSA-5687128
> [1] 15 15 15 15 15 15 15 15
> 
> $R-HSA-6796648
> [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15
> 
> $R-HSA-6803204
> [1] 15 15 15 15 15 15 15 15 15 15
> 
> $R-HSA-6803205
> [1] 15 15 15 15 15 15 15
> 
> $R-HSA-6803207
> [1] 15 15 15 15 15 15
> 
> $R-HSA-6803211
> [1] 15 15 15 15 15 15
> 
> $R-HSA-6803529
> [1] 15
> 
> $R-HSA-6804114
> [1] 15 15 15
> 
> $R-HSA-6804115
> [1] 15 15 15 15 15
> 
> $R-HSA-6804116
> [1] 15 15 15 15 15
> 
> $R-HSA-6804754
> [1] 15 15
> 
> $R-HSA-6804757
> [1] 15 15
> 
> $R-HSA-6804759
> [1] 15
> 
> $R-HSA-6807505
> [1] 15 15 15 15 15 15 15
> 
> $R-HSA-69895
> [1] 15
> 
> $R-HSA-73762
> [1] 15
> 
> $R-HSA-73863
> [1] 15
> 
> $R-HSA-8849473
> [1] 15
> 
> $R-HSA-8849474
> [1] 15
> 
> $R-HSA-8851708
> [1] 15
> 
> $R-HSA-5628897
> [1] 15
> 
> $R-HSA-5660668
> [1] 15
> 
> $R-HSA-5687128
> [1] 15 15
> 
> $R-HSA-6784531
> [1] 15
> 
> $R-HSA-6785470
> [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
> [26] 15 15 15 15 15 15 15 15 15 15 15 15
> 
> $R-HSA-6796648
> [1] 15 15 15 15 15 15 15 15
> 
> $R-HSA-6803529
> [1] 15 15
> 
> $R-HSA-6804116
> [1] 15
> 
> $R-HSA-6807505
> [1] 15 15 15
> 
> $R-HSA-69895
> [1] 15
> 
> $R-HSA-72165
> [1] 15 15 15
> 
> $R-HSA-72312
> [1] 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15
> [26] 15 15 15
> 
> $R-HSA-8857538
> [1] 15
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub

arcolombo commented:
well no. the problem is we expect duplicates,  because many of the same genes will be shared across Reactome IDs, so unique is not really what I will use...

arcolombo commented:
the good news is that speedSage is online!  

qsTable(qt,10)
    pathway.name log.fold.change     p.Value        FDR
7  R-MMU-2122947        3.468750 0.002669761 0.02314371
2  R-MMU-1368092      -93.600027 0.004302290 0.02314371
1  R-MMU-1368082     -112.798723 0.006731772 0.02314371
3  R-MMU-1368108      558.155378 0.008901428 0.02314371
4  R-MMU-1368110      558.155378 0.008901428 0.02314371
13 R-MMU-8851708       16.565316 0.013756060 0.02980480
11 R-MMU-5617472       -6.547641 0.101643499 0.18876650
10  R-MMU-400253        4.227772 0.554449707 0.89301845
8  R-MMU-2197563      -42.364901 0.628241180 0.89301845
9  R-MMU-2426168       51.022866 0.717026507 0.89301845

ttriche commented:
AWESOME!!  Nice job!

--t

> On May 4, 2016, at 6:30 PM, Anthony R. Colombo notifications@github.com wrote:
> 
> the good news is that speedSage is online!
> 
> qsTable(qt,10)
> pathway.name log.fold.change p.Value FDR
> 7 R-MMU-2122947 3.468750 0.002669761 0.02314371
> 2 R-MMU-1368092 -93.600027 0.004302290 0.02314371
> 1 R-MMU-1368082 -112.798723 0.006731772 0.02314371
> 3 R-MMU-1368108 558.155378 0.008901428 0.02314371
> 4 R-MMU-1368110 558.155378 0.008901428 0.02314371
> 13 R-MMU-8851708 16.565316 0.013756060 0.02980480
> 11 R-MMU-5617472 -6.547641 0.101643499 0.18876650
> 10 R-MMU-400253 4.227772 0.554449707 0.89301845
> 8 R-MMU-2197563 -42.364901 0.628241180 0.89301845
> 9 R-MMU-2426168 51.022866 0.717026507 0.89301845
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub

arcolombo commented:
although the build failed, my branch has passed,  so the travis yml is correct at my branch ...merging..
